src/task2/csvToJson.js

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+const csv = require('csvtojson');
+
+const filePath = '../../assets/task1.2/csv/';
+const csvFile = path.join(__dirname, filePath, 'nodejs-hw1-ex1.csv');
+const destinationFile = path.join(__dirname, filePath, 'output.txt');
+
+const readableStream = fs.createReadStream(csvFile);
+const writableStream = fs.createWriteStream(destinationFile);
+
+readableStream.on('error', (error) => {
+  console.error('Failed to load file due to error', error);
+});
+writableStream.on('error', (error) => {
+  console.error('Failed to write into file due to error', error);
+});
+
+const onError = (error) => {
+  console.log(error);
+  readableStream.close();
+  writableStream.close();
+};
+
+const onSuccess = () => {
+  readableStream.close();
+  writableStream.close();
+};
+
+const writeLine = (json) => {
+  writableStream.write(JSON.stringify(json) + os.EOL, 'utf-8');
+};
+
+// eslint-disable-next-line no-unused-vars
+const writeLineWithTimeout = (json, timeout = 0) => {
+  writeLine(json);
+  return new Promise((res) => {
+    setTimeout(() => res(), timeout);
+  });
+};
+
+csv()
```

Comment:
1. create function convertCsvToJson() in this file which coverts data and saves it output file.
2. export this function
3. in index.js import function and run it.
4. in convertCsvToJson() use abilities of pipes.
- read file line by line using `fs.createReadStream`
- convert lines one by one to json (using csvtojson)
- write to output

Reply:
I created separate file for this task with pipes
https://github.com/aleksei-bulgak-study/node-js-training/blob/33bed918930fd27591aa9dc9568067148e30e857/src/task2/csvToJsonPipeline.js

the only problem with pipes is that I can not use external libs like `readline` because it is not compatible with pipes.
And the main problem is that `createReadStream` reads whole data into buffer by default. I reduces its `highWaterMark`.
Based on your comment do you expect custom implementation for readable stream to read file  line-by-line?

Reply:
try to use this https://www.npmjs.com/package/split

Reply:
Thanks for suggestion. This lib works perfectly